### PR TITLE
RDK-29822 - Port Sky's HttpProxy plugin into an RDK Plugin

### PR DESCRIPTION
**Pull request by anttikyl**
### Description

Adds Sky's HttpProxy plugin (legacyPlugin style) as an rdkPlugin.

The HttpProxy plugin adds the ability to add `http_proxy` and `no_proxy` environment variables and proxy CA ceritificates to the container.

Disabled by default. Enabled with cmake flag `-DPLUGIN_HTTPPROXY`

### Test Procedure

To run test:
1. Fill in the `httpproxy` field in the `rdkPlugins` section of a container's config (see `rdkPlugins/HttpProxy/README.md` for instructions).
2. Launch container (make sure `postinstallhooksuccess` is not present)
3. Run `DobbyTool exec [container] cat /etc/ssl/certs/ca-certificates.crt`.

To confirm success:
1. Check that the environment variables are added to the container config as expected in the `process.env` section.
2. Check the container logs to confirm that the `proxyRootCACert` entry was properly prepended at the top of the output of reading `/etc/ssl/certs/ca-certificates.crt`.

### Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (doesn't fit into the above categories - e.g. documentation updates)

### Requires Bitbake Recipe changes?
- [ ] The base Bitbake recipe (`meta-rdk-ext/recipes-containers/dobby/dobby.bb`) must be modified to support the changes in this PR (beyond updating `SRC_REV`)

### CODE REVIEW

**Comment by mhughesacn** [comment]
I have fixed up the header complaint in the tool and will upvote.
Meantime, I noticed a certificate in the README file - I presume that this is OK for opensourcing?
Thanks,
Martin

**Comment by anttikyl** [comment]
@mhughesacn 
> Meantime, I noticed a certificate in the README file - I presume that this is OK for opensourcing?

It is only a small portion of a ceritificate, just as an example. I can replace it with something like
 `-----BEGIN CERTIFICATE-----\n[certificate]\n-----END CERTIFICATE-----\n` 
or something similar.

**Comment by mhughesacn** [comment]
Thanks Antti.  That's fine - I just have to check if I see something like it. - Martin

File: rdkPlugins/Networking/README.md [hunk 1]
```diff
@@ -101,9 +101,9 @@ If the `dnsmasq` field is empty, Networking plugin defaults to not setting up dn
 Only usable with network types 'open' and 'nat'.
 
 ```json
-    "data": {
-        "dnsmasq": "true"
-    }
+"data": {
```
Comment (TeknoVenus) [inline rdkPlugins/Networking/README.md]: These changes don't really belong in this PR

File: rdkPlugins/Networking/source/NetworkingPlugin.cpp [hunk 2]
```diff
@@ -146,7 +146,6 @@ bool NetworkingPlugin::postInstallation()
     return true;
 }
 
-
```
Comment (TeknoVenus) [inline rdkPlugins/Networking/source/NetworkingPlugin.cpp]: These changes don't really belong in this PR

File: rdkPlugins/HttpProxy/source/HttpProxyPlugin.cpp [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+        if (!mUtils->addEnvironmentVar(mContainerConfig, noProxyEnvVar))
+        {
+            AI_LOG_ERROR_EXIT("failed to add no_proxy environment variable");
+            return false;
+        }
+    }
+
+    // add the 'httpproxy' environment var
+    char httpProxyEnvVar[256];
+    snprintf(httpProxyEnvVar, sizeof(httpProxyEnvVar), "httpproxy=http://%s:%d",
```
Comment (TeknoVenus) [inline rdkPlugins/HttpProxy/source/HttpProxyPlugin.cpp:210]: This should be `http_proxy`

**Review by TeknoVenus — CHANGES_REQUESTED** [review]

File: rdkPlugins/HttpProxy/README.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    "ignoreProxy": [
+        "*.google.com",
+        "localhost"
+    ]
+}
+```
+
+### Ignore proxy on the Dobby bridge device
+
+If `ignoreProxyOnBridge` is set to true, the dobby bridge device's address is added to the `no_proxy` environment variable in the container.
```
[the remark this answers is not in the document]
Comment (anttikyl) [reply rdkPlugins/HttpProxy/README.md:72]: The host machine's localhost is accessed via the `dobby0` interface. With this, such traffic will not be proxied.

File: rdkPlugins/HttpProxy/source/HttpProxyPlugin.cpp [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+    AI_LOG_FN_ENTRY();
+
+    // get real path of the ca-certificates
+    char hostCACertsPath[PATH_MAX];
+    realpath("/etc/ssl/certs/ca-certificates.crt", hostCACertsPath);
+
+    // add a bind mount to the ca-certificates.crt file in the container's
+    // bundle. This file is created in the preCreation hook.
+    if (!mUtils->addMount(mContainerConfig, mMountedCACertsPath, hostCACertsPath, "bind",
+                         { "bind", "rec", "ro" }))
```
[the remark this answers is not in the document]
Comment (anttikyl) [reply rdkPlugins/HttpProxy/source/HttpProxyPlugin.cpp:240]: Good point, I don't think it's needed at all.

File: rdkPlugins/HttpProxy/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (TeknoVenus) [reply rdkPlugins/HttpProxy/README.md:72]: Makes sense - thanks! :)

**Review by TeknoVenus — APPROVED** [review]
LGTM

**Pull request merged**